helpers/controllers/submissionController.ts

```diff
@@ -22,34 +22,39 @@ export const createSubmission = async (
     const { challengeId, cliToken, diff, lessonId } = args
     const { id } = decode(cliToken)
     const submissionData = { diff, status: SubmissionStatus.Open }
-    const { lesson, challenge, user, ...submission } =
-      await prisma.submission.upsert({
-        where: {
-          userId_lessonId_challengeId: {
-            userId: Number(id),
-            lessonId,
-            challengeId
-          }
-        },
-        create: {
-          ...submissionData,
-          challengeId,
-          lessonId,
-          userId: Number(id)
+    const previousSubmission = await prisma.submission.findFirst({
+      where: {
+        challengeId,
+        lessonId,
+        user: {
+          id
         },
-        update: {
-          ...submissionData,
-          comments: {
-            deleteMany: {}
-          }
+        status: SubmissionStatus.Open
+      }
+    })
+    if (previousSubmission) {
+      await prisma.submission.update({
+        where: {
+          id: previousSubmission.id
         },
-        include: {
-          user: true,
-          lesson: true,
-          challenge: true
+        data: {
+          status: SubmissionStatus.Overwritten
         }
       })
-
+    }
+    const submission = await prisma.submission.create({
+      data: {
+        ...submissionData,
+        challengeId,
+        lessonId,
+        userId: Number(id)
+      },
+      include: {
+        user: true,
+        lesson: true,
+        challenge: true
+      }
```

Comment:
These includes were used only for the chat notification, because this query only expects to return the id and diff of the submission, so they can be removed until we get that back. I should've removed them on my previous mattermost deprecation PR but forgot that.

Or just leave that commented out for when someone implements the chat notification back. At least we avoid some unneded table joins for now.

Reply:
No need to keep these unused fields, so I removed them too.